drivers/adc/ad7091r8/ad7091r8.c

```diff
@@ -420,7 +431,11 @@ int ad7091r8_init(struct ad7091r8_dev **device,
 			goto err_release_reset;
 	}
 
-	/* Device powers-up in normal mode */
+	/* Set device sleep-mode */
+	ret = ad7091r8_set_sleep_mode(dev, init_param->sleep_mode);
```

Comment:
```
dev->vref_mv = init_param->vref_mv;
	if (!dev->vref_mv) {
		dev->vref_mv = 2500;
		ret = ad7091r8_set_sleep_mode(dev, AD7091R8_SLEEP_MODE_1);
		if (ret)
			goto err_release_reset;
	}
```
the code above: will be overwritten by this line, is this the intended behavior?

Reply:
Rearranged the set_sleep_mode function call above this so that user can set sleep mode / device mode using the init parameters and if vref is not set, the device will go to sleep mode 1.